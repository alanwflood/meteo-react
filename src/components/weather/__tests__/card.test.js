import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { getUnixTime } from "date-fns";
import Card from "../card";

describe("Weather Card Component", () => {
  // 1pm @ 27th September 2010
  const date = getUnixTime(new Date(2010, 8, 27, 13, 0, 0));
  const weatherData = {
    times: [1587427200, 1587438000, 1587448800, 1587459600],
    avgWeather: { icon: "10n", description: "light rain" },
    descriptions: ["light rain", "moderate rain", "light rain", "light rain"],
    icons: ["10n", "10n", "10d", "10d"],
    displayName: null,
    temps: { min: "8°C", max: "9°C", avg: "8°C", data: [1, 2, 3, 4] },
    wind: { avg: "8.38m/s", direction: "ENE", data: [1, 2, 3, 4] },
    snow: { avg: "0.00mm", data: [0, 0, 0, 0] },
    rain: { avg: "1.98mm", data: [0, 0, 0, 0] },
    humidity: { avg: "89.75%", data: [60, 70, 80, 90] },
  };
  it("Renders correctly", () => {
    const { container } = render(
      <Card isOpen weatherData={weatherData} date={date} theme="light" />
    );
    expect(container).toMatchSnapshot();
  });

  it("Opens the card when the weather button is clicked", () => {
    const { getAllByText, getByText } = render(
      <Card
        isOpen={false}
        weatherData={weatherData}
        date={date}
        theme="light"
      />
    );
    const button = getAllByText("27th September");
    fireEvent.click(button[0]);
    waitFor(() => expect(getByText("Light Rain")).toBeTruthy());
  });

  it("Shows the displayName when one is available", () => {
    const displayName = "This is a display name";
    const mockData = { ...weatherData, displayName };
    const { getByText } = render(
      <Card isOpen weatherData={mockData} date={date} theme="light" />
    );
    expect(getByText(displayName)).toBeTruthy();
  });
});
