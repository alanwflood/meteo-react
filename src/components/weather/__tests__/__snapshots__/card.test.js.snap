// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Card Component Renders correctly 1`] = `
<div>
  <div>
    <button
      class="weather-button open"
    >
      <div
        class="button-content"
      >
        <div
          class="icon weather-icon"
        >
          <test-file-stub />
        </div>
        <h2>
          27th September
          <br />
          <small>
            Monday
          </small>
        </h2>
        <div
          class="details"
        >
          <table>
            <tbody>
              <tr>
                <td>
                  Min
                </td>
                <td>
                  8°C
                </td>
              </tr>
              <tr>
                <td>
                  Max
                </td>
                <td>
                  9°C
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </button>
    <div
      class="ReactCollapse--collapse"
      style="height: auto; overflow: initial;"
    >
      <div
        class="ReactCollapse--content"
      >
        <div
          class="weather-card"
        >
          <section
            class="header"
          >
            <span>
              27th September
            </span>
          </section>
          <section
            class="content"
          >
            <div
              class="weather-details"
            >
              <div>
                <div
                  class="icon weather-icon"
                >
                  <test-file-stub />
                </div>
                <h3>
                  Light rain
                </h3>
              </div>
              <table>
                <tbody>
                  <tr>
                    <td>
                      Min Temperature
                      :
                    </td>
                    <td>
                      8°C
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Max Temperature
                      :
                    </td>
                    <td>
                      9°C
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Avg Temperature
                      :
                    </td>
                    <td>
                      8°C
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Avg Humidity
                      :
                    </td>
                    <td>
                      89.75%
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Avg Windspeed
                      :
                    </td>
                    <td>
                      8.38m/s
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Direction
                      :
                    </td>
                    <td>
                      ENE
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="react-tabs"
                data-tabs="true"
              >
                <ul
                  class="weather-tabs"
                  role="tablist"
                >
                  <li
                    aria-controls="react-tabs-1"
                    aria-disabled="false"
                    aria-selected="true"
                    class="weather-tab rainfall active"
                    id="react-tabs-0"
                    role="tab"
                    tabindex="0"
                  >
                    Rainfall
                  </li>
                  <li
                    aria-controls="react-tabs-3"
                    aria-disabled="false"
                    aria-selected="false"
                    class="weather-tab temperature"
                    id="react-tabs-2"
                    role="tab"
                  >
                    Temperature
                  </li>
                  <li
                    aria-controls="react-tabs-5"
                    aria-disabled="false"
                    aria-selected="false"
                    class="weather-tab windspeed"
                    id="react-tabs-4"
                    role="tab"
                  >
                    Windspeed
                  </li>
                  <li
                    aria-controls="react-tabs-7"
                    aria-disabled="false"
                    aria-selected="false"
                    class="weather-tab humidity"
                    id="react-tabs-6"
                    role="tab"
                  >
                    Humidity
                  </li>
                </ul>
                <div
                  aria-labelledby="react-tabs-0"
                  class="react-tabs__tab-panel react-tabs__tab-panel--selected"
                  id="react-tabs-1"
                  role="tabpanel"
                >
                  <div
                    class="weather-chart"
                  >
                    <svg
                      height="500"
                      viewBox="0 0 960 500"
                      width="960"
                    >
                      <g
                        transform="translate(50,20)"
                      >
                        <g
                          fill="none"
                          font-family="sans-serif"
                          font-size="10"
                          text-anchor="middle"
                          transform="translate(0, 450)"
                        >
                          <path
                            class="domain"
                            d="M0.5,6V0.5H890.5V6"
                            stroke="currentColor"
                          />
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(128.5,0)"
                          >
                            <line
                              stroke="currentColor"
                              y2="6"
                            />
                            <text
                              dy="0.71em"
                              fill="currentColor"
                              font-size="1.78em"
                              y="9"
                            >
                              1AM
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(339.5,0)"
                          >
                            <line
                              stroke="currentColor"
                              y2="6"
                            />
                            <text
                              dy="0.71em"
                              fill="currentColor"
                              font-size="1.78em"
                              y="9"
                            >
                              4AM
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(550.5,0)"
                          >
                            <line
                              stroke="currentColor"
                              y2="6"
                            />
                            <text
                              dy="0.71em"
                              fill="currentColor"
                              font-size="1.78em"
                              y="9"
                            >
                              7AM
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(761.5,0)"
                          >
                            <line
                              stroke="currentColor"
                              y2="6"
                            />
                            <text
                              dy="0.71em"
                              fill="currentColor"
                              font-size="1.78em"
                              y="9"
                            >
                              10AM
                            </text>
                          </g>
                        </g>
                        <g
                          fill="none"
                          font-family="sans-serif"
                          font-size="10"
                          text-anchor="end"
                        >
                          <path
                            class="domain"
                            d="M-6,450.5H0.5V0.5H-6"
                            stroke="currentColor"
                          />
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,450.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              0mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,405.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              0.5mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,360.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              1mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,315.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              1.5mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,270.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              2mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,225.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              2.5mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,180.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              3mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,135.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              3.5mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,90.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              4mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,45.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              4.5mm
                            </text>
                          </g>
                          <g
                            class="tick"
                            opacity="1"
                            transform="translate(0,0.5)"
                          >
                            <line
                              stroke="currentColor"
                              x2="-6"
                            />
                            <text
                              dy="0.32em"
                              fill="currentColor"
                              x="-9"
                            >
                              5mm
                            </text>
                          </g>
                        </g>
                        <rect
                          class="bar"
                          fill="#7ce4fe"
                          height="0"
                          width="169"
                          x="44"
                          y="450"
                        />
                        <rect
                          class="bar"
                          fill="#7ce4fe"
                          height="0"
                          width="169"
                          x="255"
                          y="450"
                        />
                        <rect
                          class="bar"
                          fill="#7ce4fe"
                          height="0"
                          width="169"
                          x="466"
                          y="450"
                        />
                        <rect
                          class="bar"
                          fill="#7ce4fe"
                          height="0"
                          width="169"
                          x="677"
                          y="450"
                        />
                      </g>
                    </svg>
                  </div>
                </div>
                <div
                  aria-labelledby="react-tabs-2"
                  class="react-tabs__tab-panel"
                  id="react-tabs-3"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="react-tabs-4"
                  class="react-tabs__tab-panel"
                  id="react-tabs-5"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="react-tabs-6"
                  class="react-tabs__tab-panel"
                  id="react-tabs-7"
                  role="tabpanel"
                />
              </div>
            </div>
          </section>
          <section>
            <h3
              style="text-align: center;"
            >
              Timeline
            </h3>
            <div
              class="timeline"
            >
              <div
                class="weather-entry"
              >
                <div
                  class="time"
                >
                  1:00AM
                </div>
                <div
                  class="description"
                >
                  <div
                    class="icon weather-icon"
                  >
                    <test-file-stub />
                  </div>
                </div>
              </div>
              <div
                class="weather-entry"
              >
                <div
                  class="time"
                >
                  4:00AM
                </div>
                <div
                  class="description"
                >
                  <div
                    class="icon weather-icon"
                  >
                    <test-file-stub />
                  </div>
                </div>
              </div>
              <div
                class="weather-entry"
              >
                <div
                  class="time"
                >
                  7:00AM
                </div>
                <div
                  class="description"
                >
                  <div
                    class="icon weather-icon"
                  >
                    <test-file-stub />
                  </div>
                </div>
              </div>
              <div
                class="weather-entry"
              >
                <div
                  class="time"
                >
                  10:00AM
                </div>
                <div
                  class="description"
                >
                  <div
                    class="icon weather-icon"
                  >
                    <test-file-stub />
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
